### ISSUE TITLE
[BUG] Cannot parse original site URL when running Acceptance test

### ISSUE THREAD

**Issue by dingo-d** [lucatume/wp-browser#419]
**Environment**
OS: MacOS 10.15.5
PHP version: 7.3.18
Installed Codeception version: 4.1.5
Installed wp-browser version: dev-feat/wptestcase-isolation
WordPress version: 5.4.1.
Local development environment: VVV
WordPress structure and management: default

**Can you perform the test manually?**
Site works when I go to the URL fine, I can log in and all

**Codeception configuration file**  
Paste, in a fenced YAML block, the content of your Codeception configuration file; remove any sensitive data!  

```yaml
paths:
    tests: tests
    output: tests/_output
    data: tests/_data
    support: tests/_support
    envs: tests/_envs
actor_suffix: Tester
extensions:
    enabled:
        - Codeception\Extension\RunFailed
    commands:
        - Codeception\Command\GenerateWPUnit
        - Codeception\Command\GenerateWPRestApi
        - Codeception\Command\GenerateWPRestController
        - Codeception\Command\GenerateWPRestPostTypeController
        - Codeception\Command\GenerateWPAjax
        - Codeception\Command\GenerateWPCanonical
        - Codeception\Command\GenerateWPXMLRPC
params:
    - .env.testing
coverage:
    enabled: true
    include:
        - src/*
    exclude:
        - src/Core/CompiledContainer.php
        - src/Old/*
    low_limit: 30
    high_limit: 75
settings:
    colors: true
    memory_limit: 2048M
```

**Suite configuration file**
Paste, in a fenced YAML block, the content of the suite configuration file; remove any sensitive data!

```yaml
# Codeception Test Suite Configuration
#
# Suite for acceptance tests.
# Perform tests in browser using the WPWebDriver or WPBrowser.
# Use WPDb to set up your initial database fixture.
# If you need both WPWebDriver and WPBrowser tests - create a separate suite.

actor: AcceptanceTester
modules:
    enabled:
        - WPDb
        - WPWebDriver
        - \Helper\Acceptance
    config:
        WPDb:
            dsn: 'mysql:host=%TEST_SITE_DB_HOST%;dbname=%TEST_SITE_DB_NAME%'
            user: '%TEST_SITE_DB_USER%'
            password: '%TEST_SITE_DB_PASSWORD%'
            dump: 'tests/_data/dump.sql'
            #import the dump before the tests; this means the test site database will be repopulated before the tests.
            populate: true
            # re-import the dump between tests; this means the test site database will be repopulated between the tests.
            cleanup: true
            waitlock: 10
            url: '%TEST_SITE_WP_URL%'
#            originalUrl: '%TEST_SITE_WP_URL%'
            urlReplacement: true #replace the hardcoded dump URL with the one above
            tablePrefix: '%TEST_SITE_TABLE_PREFIX%'
        WPWebDriver:
            originalUrl: '%TEST_SITE_WP_URL%'
            url: '%TEST_SITE_WP_URL%'
            adminUsername: '%TEST_SITE_ADMIN_USERNAME%'
            adminPassword: '%TEST_SITE_ADMIN_PASSWORD%'
            adminPath: '%TEST_SITE_WP_ADMIN_PATH%'
            browser: chrome
            host: localhost
            port: 9515
            window_size: false #disabled for Chrome driver
            capabilities:
                # Used in more recent releases of Selenium.
                "goog:chromeOptions":
                    args: ["--no-sandbox", "--headless", "--disable-gpu", "--user-agent=wp-browser"]
                # Support the old format for back-compatibility purposes.
                "chromeOptions":
                    args: ["--no-sandbox", "--headless", "--disable-gpu", "--user-agent=wp-browser"]
```

**Describe the bug**
When I uncomment the `originalUrl` config in the WPDb module I get 

```bash
In DbDump.php line 99:

  [Symfony\Component\Yaml\Exception\DumpException]
  Could not find, or could not parse, the original site URL; you can set the
  "originalUrl" parameter in the module configuration to skip this step and f
  ix this error.
```

**Output**
If applicable paste here the output of the command that's causing the issue.

Running `/vendor/bin/codecept run Acceptance -vvv` gives this:

```bash
Reading ./composer.json
Loading config file /Users/denis.zoljom/.composer/config.json
Loading config file /Users/denis.zoljom/.composer/auth.json
Loading config file ./composer.json
Checked CA file /usr/local/etc/openssl/cert.pem: valid
Executing command (/Users/denis.zoljom/vagrant-local/www/infinum/my-project/public_html/wp-content/plugins/my-plugin): git branch --no-color --no-abbrev -v
Reading /Users/denis.zoljom/.composer/composer.json
Loading config file /Users/denis.zoljom/.composer/config.json
Loading config file /Users/denis.zoljom/.composer/auth.json
Loading config file /Users/denis.zoljom/.composer/composer.json
Loading config file /Users/denis.zoljom/.composer/auth.json
Reading /Users/denis.zoljom/.composer/auth.json
Reading /Users/denis.zoljom/vagrant-local/www/infinum/my-project/public_html/wp-content/plugins/my-plugin/vendor/composer/installed.json
Reading /Users/denis.zoljom/.composer/vendor/composer/installed.json
Loading plugin Dealerdirect\Composer\Plugin\Installers\PHPCodeSniffer\Plugin
Loading plugin cweagans\Composer\Patches
Running 1.10.6 (2020-05-06 10:28:10) with PHP 7.3.18 on Darwin / 19.5.0
> test:acceptance: @php ./vendor/bin/codecept run Acceptance --ansi '-vvv'
Executing command (CWD): '/usr/local/Cellar/valet-php@7.3/7.3.18_1/bin/php' -d allow_url_fopen='1' -d disable_functions='' -d memory_limit='1536M' ./vendor/bin/codecept run Acceptance --ansi '-vvv'
Codeception PHP Testing Framework v4.1.5
Powered by PHPUnit 8.5.5 by Sebastian Bergmann and contributors.
Running with seed:


  [Connecting To Db] {"config":{"tablePrefix":"wp_","populate":true,"cleanup":true,"reconnect":false,"dump":"tests/_data/dump.sql","populator":null,"urlReplacement":true,"originalUrl":null,"waitlock":10,"dsn":"mysql:host=vvv.test;dbname=wordpresstest","user":"external","password":"external","url":"https://myproject.byinfinum.test/"},"options":[]}
  [Db] Connected to default wordpresstest

In DbDump.php line 99:

  [Symfony\Component\Yaml\Exception\DumpException]
  Could not find, or could not parse, the original site URL; you can set the
  "originalUrl" parameter in the module configuration to skip this step and f
  ix this error.


Exception trace:
  at /Users/denis.zoljom/vagrant-local/www/infinum/my-project/public_html/wp-content/plugins/my-plugin/vendor/lucatume/wp-browser/src/tad/WPBrowser/Module/Support/DbDump.php:99
 tad\WPBrowser\Module\Support\DbDump->replaceSiteDomainInSqlString() at /Users/denis.zoljom/vagrant-local/www/infinum/my-project/public_html/wp-content/plugins/my-plugin/vendor/lucatume/wp-browser/src/tad/WPBrowser/Module/Support/DbDump.php:51
 tad\WPBrowser\Module\Support\DbDump->replaceSiteDomainInSqlArray() at /Users/denis.zoljom/vagrant-local/www/infinum/my-project/public_html/wp-content/plugins/my-plugin/vendor/lucatume/wp-browser/src/Codeception/Module/WPDb.php:3951
 Codeception\Module\WPDb->_replaceUrlInDump() at /Users/denis.zoljom/vagrant-local/www/infinum/my-project/public_html/wp-content/plugins/my-plugin/vendor/lucatume/wp-browser/src/Codeception/Module/WPDb.php:3988
 Codeception\Module\WPDb->loadDumpUsingDriver() at /Users/denis.zoljom/vagrant-local/www/infinum/my-project/public_html/wp-content/plugins/my-plugin/vendor/codeception/module-db/src/Codeception/Module/Db.php:682
 Codeception\Module\Db->_loadDump() at /Users/denis.zoljom/vagrant-local/www/infinum/my-project/public_html/wp-content/plugins/my-plugin/vendor/lucatume/wp-browser/src/Codeception/Module/WPDb.php:3996
 Codeception\Module\WPDb->_loadDump() at /Users/denis.zoljom/vagrant-local/www/infinum/my-project/public_html/wp-content/plugins/my-plugin/vendor/codeception/module-db/src/Codeception/Module/Db.php:297
 Codeception\Module\Db->populateDatabases() at /Users/denis.zoljom/vagrant-local/www/infinum/my-project/public_html/wp-content/plugins/my-plugin/vendor/codeception/module-db/src/Codeception/Module/Db.php:449
 Codeception\Module\Db->_beforeSuite() at /Users/denis.zoljom/vagrant-local/www/infinum/my-project/public_html/wp-content/plugins/my-plugin/vendor/lucatume/wp-browser/src/Codeception/Module/WPDb.php:4062
 Codeception\Module\WPDb->_beforeSuite() at /Users/denis.zoljom/vagrant-local/www/infinum/my-project/public_html/wp-content/plugins/my-plugin/vendor/codeception/codeception/src/Codeception/Subscriber/Module.php:38
 Codeception\Subscriber\Module->beforeSuite() at /Users/denis.zoljom/vagrant-local/www/infinum/my-project/public_html/wp-content/plugins/my-plugin/vendor/symfony/event-dispatcher/EventDispatcher.php:230
 Symfony\Component\EventDispatcher\EventDispatcher->callListeners() at /Users/denis.zoljom/vagrant-local/www/infinum/my-project/public_html/wp-content/plugins/my-plugin/vendor/symfony/event-dispatcher/EventDispatcher.php:59
 Symfony\Component\EventDispatcher\EventDispatcher->dispatch() at /Users/denis.zoljom/vagrant-local/www/infinum/my-project/public_html/wp-content/plugins/my-plugin/vendor/codeception/codeception/src/Codeception/Event/DispatcherWrapper.php:22
 Codeception\SuiteManager->dispatch() at /Users/denis.zoljom/vagrant-local/www/infinum/my-project/public_html/wp-content/plugins/my-plugin/vendor/codeception/codeception/src/Codeception/SuiteManager.php:159
 Codeception\SuiteManager->run() at /Users/denis.zoljom/vagrant-local/www/infinum/my-project/public_html/wp-content/plugins/my-plugin/vendor/codeception/codeception/src/Codeception/Codecept.php:196
 Codeception\Codecept->runSuite() at /Users/denis.zoljom/vagrant-local/www/infinum/my-project/public_html/wp-content/plugins/my-plugin/vendor/codeception/codeception/src/Codeception/Codecept.php:163
 Codeception\Codecept->run() at /Users/denis.zoljom/vagrant-local/www/infinum/my-project/public_html/wp-content/plugins/my-plugin/vendor/codeception/codeception/src/Codeception/Command/Run.php:503
 Codeception\Command\Run->runSuites() at /Users/denis.zoljom/vagrant-local/www/infinum/my-project/public_html/wp-content/plugins/my-plugin/vendor/codeception/codeception/src/Codeception/Command/Run.php:397
 Codeception\Command\Run->execute() at /Users/denis.zoljom/vagrant-local/www/infinum/my-project/public_html/wp-content/plugins/my-plugin/vendor/symfony/console/Command/Command.php:258



COMMAND DID NOT FINISH PROPERLY.
  [Db] Disconnected from default
 Symfony\Component\Console\Command\Command->run() at /Users/denis.zoljom/vagrant-local/www/infinum/my-project/public_html/wp-content/plugins/my-plugin/vendor/symfony/console/Application.php:911
 Symfony\Component\Console\Application->doRunCommand() at /Users/denis.zoljom/vagrant-local/www/infinum/my-project/public_html/wp-content/plugins/my-plugin/vendor/symfony/console/Application.php:264
 Symfony\Component\Console\Application->doRun() at /Users/denis.zoljom/vagrant-local/www/infinum/my-project/public_html/wp-content/plugins/my-plugin/vendor/symfony/console/Application.php:140
 Symfony\Component\Console\Application->run() at /Users/denis.zoljom/vagrant-local/www/infinum/my-project/public_html/wp-content/plugins/my-plugin/vendor/codeception/codeception/src/Codeception/Application.php:117
 Codeception\Application->run() at /Users/denis.zoljom/vagrant-local/www/infinum/my-project/public_html/wp-content/plugins/my-plugin/vendor/codeception/codeception/app.php:46
 {closure}() at /Users/denis.zoljom/vagrant-local/www/infinum/my-project/public_html/wp-content/plugins/my-plugin/vendor/codeception/codeception/app.php:47
 require() at /Users/denis.zoljom/vagrant-local/www/infinum/my-project/public_html/wp-content/plugins/my-plugin/vendor/codeception/codeception/codecept:7

run [-o|--override OVERRIDE] [-e|--ext EXT] [--report] [--html [HTML]] [--xml [XML]] [--phpunit-xml [PHPUNIT-XML]] [--tap [TAP]] [--json [JSON]] [--colors] [--no-colors] [--silent] [--steps] [-d|--debug] [--bootstrap [BOOTSTRAP]] [--no-redirect] [--coverage [COVERAGE]] [--coverage-html [COVERAGE-HTML]] [--coverage-xml [COVERAGE-XML]] [--coverage-text [COVERAGE-TEXT]] [--coverage-crap4j [COVERAGE-CRAP4J]] [--coverage-phpunit [COVERAGE-PHPUNIT]] [--no-exit] [-g|--group GROUP] [-s|--skip SKIP] [-x|--skip-group SKIP-GROUP] [--env ENV] [-f|--fail-fast] [--no-rebuild] [--seed SEED] [--no-artifacts] [--] [<suite> [<test>]]
```

When I put the `urlReplacement` then I don't have that issue, but the test will timeout when I try to log in as admin `$I->loginAsAdmin()`. I tried doing 

```php
$I->amOnPage( '/wp-admin/' );
$I->makeScreenshot();
```

but the screenshot is blank. Which is weird.

The timeout error is :

```bash
Test  tests/Acceptance/Activation/PluginActivationCest.php:activatePluginSuccessfully

  [Facebook\WebDriver\Exception\TimeoutException]
```

Weird thing is that, on another plugin with 99% same setup, everything is working just fine. My acceptance tests pass, and on failure I can see the wp-admin page. So I'm not sure what am I missing or doing wrong 🤷‍♂️

**Comment by lucatume** [comment]
The issue would come to be only if the dump loaded by the `WPDb` module is empty or does not contain a statement like `INSERT INTO ...options... (home|siteurl)`.

Could you check the file `tests/_data/dump.sql` and let me know what is in there?

**Comment by dingo-d** [comment]
Yeah, I have it like 

```sql
INSERT INTO `wp_options` VALUES (1,'siteurl','https://myproject.byinfinum.test','yes'),(2,'home','https://myproject.byinfinum.test','yes')...
```

Which was strange. I dumped the SQL using the `mysqldump` (thought that it was an issue when you dump the DB using phpMyAdmin, but it's not).

**Comment by dingo-d** [comment]
Another weird thing is, I tried runnnig functional tests (Cest format) and I tried using `$I->sendAjaxPostRequest`, but I got an error:

```bash
[RuntimeException] Call to undefined method FunctionalTester::sendAjaxPostRequest
```

**Comment by Luc45** [comment]
@dingo-d 

That looks accurate.

[WPBrowser](https://github.com/lucatume/wp-browser/blob/master/src/Codeception/Module/WPBrowser.php) extends [PhpBrowser](https://github.com/Codeception/module-phpbrowser/blob/master/src/Codeception/Module/PhpBrowser.php) which extends [InnerBrowser](https://github.com/Codeception/lib-innerbrowser/blob/master/src/Codeception/Lib/InnerBrowser.php).

The method `sendAjaxPostRequest` is defined in `InnerBrowser`: https://github.com/Codeception/lib-innerbrowser/blob/master/src/Codeception/Lib/InnerBrowser.php#L1268-L1291

What I think might be confusing you is that there is a process that happens behind the scenes every time you run a test, which is `codecept build`.

It generates the Actor classes, such as `$I`, based on the active modules for that suite. If you change the modules of a suite but didn't run any test yet, your IDE will be reading from the `$I` actor class that was generated with the modules from on your previous test run.

To avoid that lesser inconvenience, you might run `codecept build` manually after tweaking your modules so that your IDE can catch up to what changed without having to run a test.

**Comment by Luc45** [comment]
About the issue with the dump itself, I don't know, I've never faced that error.

I've already faced problems when my `dump.sql` was wrapped in transactions (which is a default checkbox when you export through PhpMyAdmin). You might try to tweak your `dump.sql` a little bit and see if that helps.

**Comment by dingo-d** [comment]
I'll try this. I did have some issues with content from the dump, so it could be that the dump is causing this. It's strange behavior because I cannot reproduce it in the other plugin where everything works fine.

I even tried using that other `dump.sql` to see if I'd get the issue, but it still returned white screenshot.

**Comment by lucatume** [comment]
@dingo-d As a first step to debug this I will test the dump line you provided against the regular expression that is, currently, trying to infer the `home` or `siteurl` from the options table and see if that might be the issue.

Will post updates here.

**Comment by dingo-d** [comment]
Ok, this is weird. I'm getting timeout on my other plugin now as well. Could this be related to VVV?

**Comment by lucatume** [comment]
@dingo-d I've tested the SQL fragment you provided against the current parser, and the URL is correctly picked up.
Are you still, consistently, having this issue?

> When I put the urlReplacement then I don't have that issue, but the test will timeout when I try to log in as admin $I->loginAsAdmin()

A blank screenshot means a 404 in the context of acceptance tests.
If the dump contains a URL that is not the one used to access the installation during tests, then that is the cause.

Did you try to set `originalUrl` in the `WPDb` configuration?
See [here for more information](https://wpbrowser.wptestkit.dev/modules/wpdb#configuration).

Could you, please, post the `--debug` output of the command when  you set the `originalUrl` in the `WPDb` module configuration?

Let me know,

**Comment by dingo-d** [comment]
Yeah, if I remove the `urlReplacement` I still get the same error 

```bash
Could not find, or could not parse, the original site URL; you can set the
  "originalUrl" parameter in the module configuration to skip this step and f
  ix this error.
```

> A blank screenshot means a 404 in the context of acceptance tests.
If the dump contains a URL that is not the one used to access the installation during tests, then that is the cause.

I checked my dump, and there were places where URL had http instead of https. I've changed that, but it still times out.

> Did you try to set originalUrl in the WPDb configuration?
See here for more information.

Yes, it only works if I put the `originalUrl` in the `WPDb` module config.

I went into `DbDump.php` and checked the `getOriginalUrlFromSqlString()` method. When I print out the `$matches` it just shows an empty array. Weird thing is that when I use regex101 to check the match I do get it

![Screenshot 2020-06-09 at 07 51 26](https://user-images.githubusercontent.com/8638515/84110908-11a92b80-aa26-11ea-8c1c-10d359cbd678.png)

So I dug up a bit and noticed that I have tons of things in my postmeta so I removed that and lo and behold it now works without using the `originalUrl` set.

So I guess that there is a limit to how big the dump is (my SQL dump file is like 2.9MB which is a lot). Maybe there is a limit to what PHP can process.

The debug statement on my TimeoutException is

```bash
composer test:acceptance -- --debug
> @php ./vendor/bin/codecept run Acceptance --ansi '--debug'
Codeception PHP Testing Framework v4.1.6
Powered by PHPUnit 8.5.5 by Sebastian Bergmann and contributors.
Running with seed:


  [Connecting To Db] {"config":{"tablePrefix":"wp_","populate":true,"cleanup":true,"reconnect":false,"dump":"tests/_data/dump.sql","populator":null,"urlReplacement":true,"originalUrl":null,"waitlock":10,"dsn":"mysql:host=vvv.test;dbname=wordpresstests","user":"external","password":"external","url":"https://myproject.byinfinum.test/"},"options":[]}
  [Db] Connected to default wordpresstests
  [Query] INSERT INTO `wp_options` (`option_name`, `option_value`, `autoload`) VALUES (?, ?, ?)
  [Parameters] ["admin_email_lifespan",2533080438,"yes"]
  [Query] INSERT INTO `wp_options` (`option_name`, `option_value`, `autoload`) VALUES (?, ?, ?)
  [Parameters] ["_transient_doing_cron",1591683260,"yes"]

Acceptance Tests (1) -----------------------------------------------------------
Modules: WPDb, WPWebDriver, \Helper\Acceptance
--------------------------------------------------------------------------------
PluginActivationCept:
Signature: PluginActivationCept
Test: tests/Acceptance/Activation/PluginActivationCept.php
Scenario --
  [Query] INSERT INTO `wp_options` (`option_name`, `option_value`, `autoload`) VALUES (?, ?, ?)
  [Parameters] ["admin_email_lifespan",2533080438,"yes"]
  [Query] INSERT INTO `wp_options` (`option_name`, `option_value`, `autoload`) VALUES (?, ?, ?)
  [Parameters] ["_transient_doing_cron",1591683265,"yes"]
 I am on page "/wp-admin/"
  [GET] https://myproject.byinfinum.test/wp-admin/
 I make screenshot
  [Screenshot Saved] file:///Users/denis.zoljom/vagrant-local/www/infinum/my-project/public_html/wp-content/plugins/my-plugin/tests/_output/debug/2020-06-09_06-05-26_5edf26a61ec4e.png
 I login as admin
  Trying to login, attempt 0/5...
  [GET] https://myproject.byinfinum.test/wp-login.php
  Screenshot and page source were saved into '/Users/denis.zoljom/vagrant-local/www/infinum/my-project/public_html/wp-content/plugins/my-plugin/tests/_output/' dir
 ERROR

--------------------------------------------------------------------------------


Time: 17.64 seconds, Memory: 14.00 MB

There was 1 error:

---------
1) PluginActivationCept:
 Test  tests/Acceptance/Activation/PluginActivationCept.php

  [Facebook\WebDriver\Exception\TimeoutException]


Scenario Steps:

 3. $I->loginAsAdmin() at tests/Acceptance/Activation/PluginActivationCept.php:6
 2. $I->makeScreenshot() at tests/Acceptance/Activation/PluginActivationCept.php:5
 1. $I->amOnPage("/wp-admin/") at tests/Acceptance/Activation/PluginActivationCept.php:4

#1  /Users/denis.zoljom/vagrant-local/www/infinum/my-project/public_html/wp-content/plugins/my-plugin/vendor/php-webdriver/webdriver/lib/WebDriverWait.php:71
#2  /Users/denis.zoljom/vagrant-local/www/infinum/my-project/public_html/wp-content/plugins/my-plugin/vendor/lucatume/wp-browser/src/Codeception/Module/WPWebDriver.php:82
#3  /Users/denis.zoljom/vagrant-local/www/infinum/my-project/public_html/wp-content/plugins/my-plugin/vendor/lucatume/wp-browser/src/Codeception/Module/WPWebDriver.php:45
#4  Codeception\Module\WPWebDriver->loginAsAdmin
#5  /Users/denis.zoljom/vagrant-local/www/infinum/my-project/public_html/wp-content/plugins/my-plugin/tests/_support/_generated/AcceptanceTesterActions.php:4038
#6  /Users/denis.zoljom/vagrant-local/www/infinum/my-project/public_html/wp-content/plugins/my-plugin/tests/Acceptance/Activation/PluginActivationCept.php:6
Artifacts:

Png: /Users/denis.zoljom/vagrant-local/www/infinum/my-project/public_html/wp-content/plugins/my-plugin/tests/_output/PluginActivationCept.fail.png
Html: /Users/denis.zoljom/vagrant-local/www/infinum/my-project/public_html/wp-content/plugins/my-plugin/tests/_output/PluginActivationCept.fail.html

ERRORS!
Tests: 1, Assertions: 0, Errors: 1.
  [Db] Disconnected from default
Script @php ./vendor/bin/codecept run Acceptance --ansi handling the test:acceptance event returned with error code 1
```

**Comment by Luc45** [comment]
> So I guess that there is a limit to how big the dump is (my SQL dump file is like 2.9MB which is a lot). Maybe there is a limit to what PHP can process.

Just an observation, I've worked on projects where the dump would be like 40MB. It took a while, but it worked, both with the populator and without it.

You said you isolated the issue to your postmeta table? Can you isolate it even more, leave just a few records and see exactly what might be triggering it? Eg: To find out if this is coming from the structure or something in the data?

**Comment by dingo-d** [comment]
I'll try to dig deeper into it. But the timeout started to occur with another plugin with a smaller dump. So it could be something system-related 🤷‍♂️

**Comment by dingo-d** [comment]
So, when I remove the `originalUrl: '%TEST_SITE_WP_URL%'` from the `WPDb` module, I get error, unless I remove some parts of my `wp_postmeta` in the dump. Which is weird, and could mean that I have a malformed dump.

When I add this, I get the `Facebook\WebDriver\Exception\TimeoutException` no matter what I do with the dump. The URL in the dump is correct.

Furthermore, when I changed from `WPWebDriver` to `WPBrowser` the test passes. And I can make `makeHtmlSnapshot` and see everything executed correctly.

**Comment by Luc45** [comment]
Ah, this timeout is coming from the WebDriver, not the dump.

Do you have your `wp-config.php` pointing to the test database?

**Comment by dingo-d** [comment]
No, my test database is some empty DB I use, different from the one I'm using from the `wp-config.php`. Should these be the same? 

The output I'm getting is the white screen with nothing on it.

**Comment by lucatume** [comment]
> No, my test database is some empty DB I use, different from the one I'm using from the wp-config.php. Should these be the same?

I think there might be an issue here where the WordPress installation uses database A (in the `wp-config.php` file), but the tests use, in the `WPDb` module configuration, database B.
@dingo-d  did you see [this?](https://wpbrowser.wptestkit.dev/tutorials/automatically-change-db-in-tests)?

**Comment by Luc45** [comment]
> EDIT: I've changed the .env.testing to point to the same DB as the one in my wp-config.php and it still timed out 🤷‍♂️

Noooo manz don't do this, this will import the test database on top of your local development database 🤦‍♂️ 

Do you want to hang to talk about this stuff? I could explain it better to you in a call, if you'd like. Let me know.

**Comment by dingo-d** [comment]
> > No, my test database is some empty DB I use, different from the one I'm using from the wp-config.php. Should these be the same?
> 
> I think there might be an issue here where the WordPress installation uses database A (in the `wp-config.php` file), but the tests use, in the `WPDb` module configuration, database B.
> @dingo-d did you see [this](https://wpbrowser.wptestkit.dev/tutorials/automatically-change-db-in-tests)?

Yeah, I've added this but still times out.

> > EDIT: I've changed the .env.testing to point to the same DB as the one in my wp-config.php and it still timed out 🤷‍♂️
> 
> Noooo man, don't do this, this will import the test database on top of your local development database 🤦‍♂️
> 
> Do you want to hang to talk about this stuff? I could explain it better to you in a call, if you'd like. Let me know.

I'm aware of the databases, I usually use a separate one for tests from the one I use in the development, don't worry 😄  It's also easily recoverable data 😄 

I'm up for a call, but maybe tomorrow, thanks for helping out 👍

**Comment by Luc45** [comment]
> I'm up for a call, but maybe tomorrow, thanks for helping out +1

Feel free to schedule a hangout with me tomorrow lucasfbustamante@gmail.com I'm on GMT-3 and I'll be around all day, mostly.

**Comment by dingo-d** [comment]
I've send an invitation, let me know if the time suits you, if not we can move it around a bit 🙂

**Comment by dingo-d** [comment]
Ok, so @Luc45 you're a lifesaver. After battling with valet+ for about an hour or so, I gave up and decided to just use vvv.

First, my port was wrong in my `WPWebDriver` config. I'm starting my selenium using composer script and there it's `selenium-server -port 4444`, and I had port opened to 9515 instead of 4444 (although I thought that you need to have chromedriver running as well).

Then I totally didn't realize I can mess with chrome config. Removing `--headless` flag brings up the browser so I could actually see what was wrong. And it was the certificate.

My URL (on the plugin where I tested this) is `https://dev.wordpress.test`, and I got certificate warning. So I added `"allow-insecure-localhost"` and `"--ignore-certificate-errors"` flags to my `"goog:chromeOptions"` and `"chromeOptions"` in the `Acceptance.suite.yml`.

Once this is set, the certificate errors are ignored and I can use HTTPS (which I kinda need since I'm testing WooCommerce, and it will complain). On projects where I don't really care about HTTPS, I can just set the URL to use `http://`.

So all is well with the `wp-browser` 🎉 🎉  

It's my stupid local setup that was to blame.

Thank you guys again for all the help 🙂  If we ever meet at some WordCamp I owe you a beer 😄  🍻

**Comment by lucatume** [comment]
That's great to hear @dingo-d,

thanks for writing down what issues you found and reporting them back: this will help others with the same issues to debug them.

**Comment by hatchjaw** [comment]
I had the same issue as @dingo-d; preg_match in DbDump::getOriginalUrlFromSqlString() was failing to match the original URL. Clearing transients (`wp transient delete --all`), then re-dumping the database seemed to sort it out for me. It's not clear which transient may have been causing the problem (maybe something WooCommerce-related).

Interestingly, even with the working dump, regex101 encounters catastrophic backtracking after ".*"

![Screen Shot 2021-03-23 at 01 14 09](https://user-images.githubusercontent.com/4884512/112077809-226dfc80-8b75-11eb-8301-1c422e072594.png)

Removing the /s (dot matches newline) flag stops that happening.

Perhaps getOriginalUrlFromSqlString could be more selective about where it looks in the dump, just checking the options table rather than the whole thing. Maybe the /s flag is unnecessary too. ".*" may be too aggressive.

**Comment by lucatume** [comment]
@hatchjaw this is worth taking a second look at. 

The current solution is not quite as robust as I would like, and the best way to mitigate this is to set the `originalUrl` explicitly in the [module configuration](https://wpbrowser.wptestkit.dev/modules/wpdb).

Next time I work on the code I will spend some time reviewing this.

**Comment by hatchjaw** [comment]
Thanks @lucatume. I had some success with `originalUrl`, but the suspicious part of me wondered what else I might be overriding by setting that in config, particularly since it's not used in any of the examples I could find. Glad to know that's the recommended workaround.

**Comment by stracker-phil** [comment]
I ran into the same issue today, and after adding a few debug outputs to DbDump.php I found the problem:

The regular expression in function `getOriginalUrlFromSqlString()` due to so-called "[catastrophic backtracking](http://www.regular-expressions.info/catastrophic.html)".

### The problematic regex:

``/INSERT\s+INTO\s+`wp_options`.*'(home|siteurl)'\s*,\s*'(?<url>[^']+)'/uis``

The problem is caused by the `.*` part before `(home|siteurl)`: 

When the dump file contains the term 'home' somewhere after the wp_options data, then the regular expression will evaluate all possible matches and finally give up due to the number of characters between both 'home' terms;

I have a page called "Home" in my DB. Also, there could be all kind of other cases where 'home' occurs somewhere in posts, post_meta, terms, or user_meta (or any other table that comes after wp_options)

**Note**

I've used the Regex debugger here to process the regex step by step and the problem was very obvious :)  
https://regex101.com/ → paste the regex and the entire wp_options table dump. Then click on "Regex Debugger" in the left sidebar to simulate each step of the regex engine.

### TLDR

After adjusting the regex to be ungreedy, the problem is perfectly solved:

``/INSERT\s+INTO\s+`wp_options`.*'(home|siteurl)'\s*,\s*'(?<url>[^']+)'/Uuis`` ← added `/U`

The effect of the `/U` flag is clearly visible in the regex debugger.

**Comment by EvanShaw** [comment]
@stracker-phil 
You are a life saver.

I noticed this same problem when testing two different dump files with the same exact URL to replace. One would fail and the other succeed. The only difference was their respective size. I'm not enough of a regex guru to spot the catastrophic backtracking problem so I was completely stumped.

Thanks for fixing this!

**Comment by lucatume** [comment]
Thanks @stracker-phil  for the PR, I've wrapped it to have all the tests running on it and will release as soon as I'm able to.

**Comment by lucatume** [comment]
Follow up: [the fix is in and merged.](https://github.com/lucatume/wp-browser/releases/tag/3.0.7)
